### ISSUE TITLE
Running tests on Windows gets terminated with STATUS_STACK_OVERFLOW

### ISSUE THREAD

**Issue by oroppas** [RustPython/RustPython#2926]
### Release

```
cargo run --release -- -m test -v
    Finished release [optimized] target(s) in 3.07s
     Running `target\release\rustpython.exe -m test -v`
WARNING: The filename '@test_19540_tmp-共Ł♡ͣ�' CAN be encoded by the filesystem encoding (utf-8). Unicode filename tests may not be effective
== RustPython 3.9.0alpha (heads/master-dirty:4f0feef9, Aug 22 2021, 05:00:44) [rustc 1.54.0]
== Windows-8-6.2.9200-SP0 little-endian
== cwd: C:\Users\ryuta\AppData\Local\Temp\test_python_19540
== CPU count: 12
== encodings: locale=UTF-8, FS=utf-8
Run tests sequentially
0:00:00 [  1/215] test___future__
test_attributes (test.test___future__.FutureTest) ... ok
test_names (test.test___future__.FutureTest) ... ok

----------------------------------------------------------------------

Ran 2 tests in 0.002s

OK
0:00:00 [  2/215] test_abc -- test___future__ passed
test_works_with_init_subclass (test.test_abc.test_factory.<locals>.TestABCWithInitSubclass) ... ok
test_works_with_init_subclass (test.test_abc.test_factory.<locals>.TestABCWithInitSubclass) ... ok

...

test_readlines (test.test_io.PyBufferedReaderTest) ... ok
test_readonly_attributes (test.test_io.PyBufferedReaderTest) ... ok
test_recursive_repr (test.test_io.PyBufferedReaderTest) ...
thread 'main' has overflowed its stack
error: process didn't exit successfully: `target\release\rustpython.exe -m test -v` (exit code: 0xc00000fd, STATUS_STACK_OVERFLOW)
```

### Debug

```
cargo run -- -m test -v
    Finished dev [unoptimized + debuginfo] target(s) in 3.44s
     Running `target\debug\rustpython.exe -m test -v`

thread 'main' has overflowed its stack
error: process didn't exit successfully: `target\debug\rustpython.exe -m test -v` (exit code: 0xc00000fd, STATUS_STACK_OVERFLOW)
```

What's the best way to diagnose this?

**Comment by youknowone** [comment]
#2881 looks the same issue

**Comment by youknowone** [comment]
is this also solved by #3082?

**Comment by oroppas** [comment]
@youknowone Yes, the situation gets much improved. The tests can be run on release/debug on msvc/msys.
Thanks!
